### PR TITLE
Update sst deps

### PR DESCRIPTION
**Pull request by chuckatkins**
Fix an array of issues with SST dependencies

### CODE REVIEW

**Comment by chuckatkins** [comment]
@philip-davis @eisenhauer see if this fixes your libfabric issues

File: source/adios2/toolkit/sst/CMakeLists.txt [hunk 1]
```diff
@@ -6,11 +10,18 @@ add_library(sst
   cp/cp_common.c
   cp/ffs_marshal.c
 )
-
-if(SST_HAVE_LIBFABRIC)
-    target_sources(sst PRIVATE dp/rdma_dp.c)
+if(ADISO2_SST_HAVE_LIBFABRIC)
```
Comment (philip-davis) [inline source/adios2/toolkit/sst/CMakeLists.txt]: There's a typo in this variable name

**Comment by philip-davis** [comment]
This seems to fix the dependency issues, once the typo above is fixed. Thanks!

**Comment by chuckatkins** [comment]
Apparently this is still broken for static libs on sunmitdev.  Will hold off on merging until it can be addressed.

**Pull request merged**